rtsp.go

```diff
@@ -287,21 +290,13 @@ func (rc *rtspCamera) initH264(session *description.Session) (err error) {
 			return
 		}
 
-		for _, nalu := range au {
-			if len(nalu) < 20 {
-				// TODO(ERH): this is probably wrong, but fixes a spam issue with "no frame!"
-				continue
-			}
-			// convert NALUs into RGBA frames
-			image, err := rc.rawDecoder.decode(nalu)
-			if err != nil {
-				rc.logger.Errorf("error decoding(2) h264 rtsp stream  %s", err.Error())
-				return
-			}
-			if image != nil {
-				rc.latestFrame.Store(&image)
-			}
+		if !receivedFirstIDR && h264.IDRPresent(au) {
+			rc.logger.Debug("adding inital SPS & PPS")
+			receivedFirstIDR = true
+			au = append(initialSPSAndPPS, au...)
```

Comment:
I confirmed that if the SPS & PPS are also sent in the first IDR au (so [SPS, PPS, SPS, PPS, IDR]) that is able to be decoded into a jpeg without causing any error log spam.

Reply:
nice, that is handy! So it seems if the first IDR has different SPS and PPS from the SDP-derived ones, they will be overwritten by the ordering here.